mutiny-core/src/wallet.rs

```diff
@@ -87,6 +88,37 @@ impl MutinyWallet {
         Ok(())
     }
 
+    pub(crate) async fn insert_tx(
+        &self,
+        tx: Transaction,
+        position: ConfirmationTime,
+        block_id: Option<BlockId>,
+    ) -> Result<(), MutinyError> {
+        match position {
+            ConfirmationTime::Confirmed { .. } => {
+                // if the transaction is confirmed and we have the block id,
+                // we can insert it directly
+                if let Some(block_id) = block_id {
+                    let mut wallet = self.wallet.try_write()?;
+                    wallet.insert_checkpoint(block_id)?;
+                    wallet.insert_tx(tx, position)?;
+                } else {
+                    // if the transaction is confirmed and we don't have the block id,
+                    // we should just sync the wallet otherwise we can get an error
+                    // with the wallet being behind the blockchain
+                    self.sync().await?
```

Comment:
I'm guessing the `sync()` function then handles the inserts?

And there's no clashing or problems that we are just inserting directly into it's db?

Reply:
> I'm guessing the sync() function then handles the inserts?

Yeah it does.

> And there's no clashing or problems that we are just inserting directly into it's db?

Yeah they do a similar in memory caching thing that we do.